lib/manageiq/api/common/application_controller_mixins/request_body_validation.rb

```diff
@@ -38,6 +38,7 @@ def body_params
           # - only for HTTP POST/PATCH
           def validate_request
             return unless request.post? || request.patch?
+            return unless self.class.to_s.split('::').length > 2
```

Comment:
Is there a better way to do this?  Maybe looking at the request path and only validating if the path is in the OpenAPI document?

Reply:
Out of curiosity, do you have a use case for undocumented controllers?

Reply:
@bdunne We need the version and the path to get to the openapi document. This call fails before that.

The use case that we have is to limit the number of rescue_froms that we need to write in the controllers.
Erik was following this https://coderwall.com/p/iumbag/quickly-enable-middleware-based-custom-exception-handling-in-rails as an alternative to using rescue_froms. We have a controller call ErrorsController which is not path'ed for the openapi.

https://github.com/ManageIQ/catalog-api/blob/master/app/controllers/errors_controller.rb

Which emanates from the routes defined here
https://github.com/ManageIQ/catalog-api/blob/1ade13072b252848ace0e12e0312addb8fcdc933/config/routes.rb#L7

Reply:
I don't understand how this change means "non openapi controller".  I read it as "any controller with a namespace", which isn't the same thing.

Reply:
@Fryguy Can you please review changed it to use a class variable that can be set in non openapi controllers.